constraint_utils/constraint_basics.py

```diff
@@ -425,7 +463,46 @@ def get_proportion_observed(exome_ht: hl.Table, context_ht: hl.Table, mutation_h
-
+    """
+    Compute the expected number of variants using plateau models and coverage model.
+
+    This function sums the number of possible variants times the mutation rate for all variants, and applies the calibration
+    model separately for CpG transitions and other sites. For sites with coverage lower than the coverage cutoff, the value obtained 
+    from the previous step is multiplied by the coverage correction factor. These values are summed across the set of variants 
+    of interest to obtain the expected number of variants.
+    
+    A brief view of how to get the expected number of variants:
+    mu_agg = the number of possible variants * the mutation rate (all variants)
+    adjusted_mutation_rate = sum(plateau model slop * mu_agg + plateau model intercept) (separately for CpG transitions and other sites)
+    if 0 < coverage < coverage cutoff:
+        coverage_correction = coverage_model slope * log10(coverage) + coverage_model intercept
+        expected_variants = sum(adjusted_mutation_rate * coverage_correction)
+    else:
+        expected_variants = sum(adjusted_mutation_rate)
+    The expected_variants are summed across the set of variants of interest to obtain the final expected number of variants.
+    
+    Function adds the following annotations:
+        - variant_count - annotated by `count_variants` function
+        - adjusted_mutation_rate (including those for each population) - mutation rate adjusted by plateau model
+        - possible_variants (including those for each population)
+        - expected_variants (including those for each population)
+        - mu - sum(mu_snp * possible_variant * coverage_correction)
+        - obs_exp - observed:expected ratio
+        - constraint annotated by `annotate_constraint_groupings`
+
+    :param exome_ht: Exome site Table (output of `prepare_ht`) filtered to autosomes and pseudoautosomal regions.
+    :param context_ht: Context Table (output of `prepare_ht`) filtered to autosomes and pseudoautosomal regions.
+    :param mutation_ht: Mutation rate Table with 'mu_snp' field.
+    :param plateau_models: A linear model (output of `build_plateau_models_pop`) that calibrates mutation rate to proportion observed for high coverage exome. It includes models for CpG site, non-CpG site, and each population in `POPS`.
+    :param coverage_model: A linear model (output of `build_coverage_model`) that calibrates a given coverage level to observed:expected ratio. It's a correction factor for low coverage sites.
+    :param recompute_possible: Whether to use context Table to recompute the number of possible variants instead of using a precomputed intermediate Table if it exists. Defaults to False.
+    :param remove_from_denominator: Whether to remove alleles in context Table with coverage and allele count equal to 0, frequency larger than `af_cutoff`, and can't be found in `exome_ht`, defaults to True
```

Comment:
It's kind of wordy to include all the information. Is there any way to rephrase it?